lib/beaker/hypervisor/docker.rb

```diff
@@ -65,6 +65,19 @@ def provision
         host['docker_container'] = container
         host['docker_image'] = image
       end
+
+      hack_docker_etc_hosts @hosts, @options
+
+    end
+
+    def hack_docker_etc_hosts hosts, opts
+      etc_hosts = "127.0.0.1\tlocalhost localhost.localdomain\n"
+      hosts.each do |host|
+        etc_hosts += "#{host['docker_container'].json["NetworkSettings"]["IPAddress"]}\t#{host.name}\n"
```

Comment:
example entry of a docker instance:

```
# cat /etc/hosts
172.17.0.15     centos-6-x64
127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
```

This is with a remote docker server btw.
The hosts file looks good to me :-)